process/block/shardblock.go

```diff
@@ -742,17 +746,55 @@ func (sp *shardProcessor) restoreMetaBlockIntoPool(mapMiniBlockHashes map[string
 
 	for metaBlockHash, miniBlockHashes := range mapMetaHashMiniBlockHashes {
 		for _, miniBlockHash := range miniBlockHashes {
-			sp.processedMiniBlocks.AddMiniBlockHash(metaBlockHash, string(miniBlockHash))
+			//TODO: Check if needed, how to set the real index (metaBlock -> ShardInfo -> ShardMiniBlockHeaders -> TxCount)
+			indexOfLastTxProcessed := common.MaxIndexOfTxInMiniBlock
+			sp.processedMiniBlocks.SetProcessedMiniBlockInfo([]byte(metaBlockHash), miniBlockHash, &processedMb.ProcessedMiniBlockInfo{
+				IsFullyProcessed:       true,
+				IndexOfLastTxProcessed: indexOfLastTxProcessed,
+			})
 		}
 	}
 
 	for miniBlockHash := range mapMiniBlockHashes {
-		sp.processedMiniBlocks.RemoveMiniBlockHash(miniBlockHash)
+		miniBlockHeader := process.GetMiniBlockHeaderWithHash(headerHandler, []byte(miniBlockHash))
```

Comment:
please extract this into a method rollback processedMiniBlock with the miniBlock hash and header handler as parameters

Reply:
Done